lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,28 @@
+module Semian
+  class AtomicInteger #:nodoc:
+    attr_accessor :value
+
+    def initialize(_name, _permissions)
+      @value = 0
+    end
+
+    def increment_by(val)
+      self.value += val
+    end
+
+    def increment
+      increment_by(1)
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
+
+    def shared?
+      false
+    end
```

Comment:
Sharing tests are a good idea. A single test testing them all is not.

What I would recommend is simply using a module:

```ruby
class SysVAtomicIntegerTest < Minitest::TestCase
  include SharedAtomicIntegerTestCases

  # Add tests specific to SysVAtomicInteger here.
end
```